modules/Microsoft.Automation/automationAccounts/deploy.bicep

```diff
@@ -169,17 +172,17 @@ resource automationAccount 'Microsoft.Automation/automationAccounts@2020-01-13-p
     sku: {
       name: skuName
     }
-    encryption: {
-      identity: encryptionKeySource == 'Microsoft.Keyvault' ? {
-        userAssignedIdentity: any(encryptionUserAssignedIdentity)
-      } : null
-      keySource: encryptionKeySource
-      keyVaultProperties: encryptionKeySource == 'Microsoft.Keyvault' ? {
-        keyName: keyName
-        keyvaultUri: keyvaultUri
-        keyVersion: keyVersion
-      } : null
-    }
+    encryption: !empty(cMKKeyName) ? {
+      keySource: 'Microsoft.KeyVault'
+      identity: {
+        userAssignedIdentity: cMKUserAssignedIdentityResourceId
+      }
+      keyVaultProperties: {
+        keyName: cMKKeyName
+        keyVaultUri: cMKKeyVault.properties.vaultUri
+        keyVersion: !empty(cMKKeyVersion) ? cMKKeyVersion : last(split(cMKKeyVaultKey.properties.keyUriWithVersion, '/'))
```

Comment:
Properties look very similar to the DataFactory implementation. Could it be tested if `keyVersion: !empty(cMKKeyVersion) ? cMKKeyVersion : null` works in this case as well i.e. if also this resource automatically assumes latest if no version is provided?

Reply:
I did, and no. It is very inconsistent how the Product-Groups implemented the CMK interface. Not providing a version was usually always the first thing I tried.